### PR TITLE
SIR-200 SIR-198 finished

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/Api.Common/Domain/Accounts/Account.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                accountId,
+                brokerAccountId,
+                referenceId,
+                accountState,
+                creationDateTime,
+                blockingDateTime,
+                activationDateTime);
+        }
+
+        public void Activate()
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Domain/Accounts/Account.cs]: This shouldn't be in read-model

Comment (KonstantinRyazantsev) [reply src/Api.Common/Domain/Accounts/Account.cs]: This model is not designed as a read model at all. As we've discussed we'll refactor it once entire epic is complete

File: src/Api.Common/Persistence/AccountRepository.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        public async Task UpdateAsync(Account account)
+        {
+            await using var context = new DatabaseContext(_dbContextOptionsBuilder.Options);
+            var entity = MapToEntity(account);
+            context.Accounts.Update(entity);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task<Account> AddOrGetAsync(Account account)
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/AccountRepository.cs]: Is this method used for the read model?

File: src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs [hunk 3]
```diff
@@ -23,9 +23,10 @@ public async Task Consume(ConsumeContext<BrokerAccountActivated> context)
         {
             var message = context.Message;
 
-            var brokerAccount = await _brokerAccountRepository.GetAsync(message.BrokerAccountId);
+            var brokerAccount = await _brokerAccountRepository.GetOrDefaultAsync(message.BrokerAccountId);
 
-            if (!brokerAccount.ActivationDateTime.HasValue)
+            if (brokerAccount != null &&
```
Comment (KonstantinRyazantsev) [inline src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs]: brokerAccount should be in the DB. If it's missed then probably messages are disordered or previous message is lost. So, this message should be retried in this case and end up in the error queue eventually.

File: src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs [hunk 4]
```diff
@@ -23,9 +23,10 @@ public async Task Consume(ConsumeContext<BrokerAccountActivated> context)
...
+                !brokerAccount.ActivationDateTime.HasValue)
```
Comment (KonstantinRyazantsev) [inline src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs]: We don't need to check it here. Account can be activated several times. (blocked, then activated again). We have to have something for messages ordering. But this is a cross-cutting concern. We'll solve it later on.

File: src/Api/WebApi/AccountController.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var tenantId = await this.GetTenantIdAsync();
+
+            long.TryParse(brokerAccountId, out var brokerAccountIdLong);
+            var brokerAccount = await _brokerAccountRepository.GetOrDefaultAsync(brokerAccountIdLong);
```
Comment (KonstantinRyazantsev) [inline src/Api/WebApi/AccountController.cs]: That's a good example, where GetOrDefault fits good

File: src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs (hunk 3 above)
Comment (KonstantinRyazantsev) [reply src/Api.Worker/MessageConsumers/BrokerAccountActivatedConsumer.cs]: So we additionally need Get method in the repo.

File: src/Api/WebApi/AccountController.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+            if (brokerAccount == null)
+            {
+                ModelState.AddModelError(nameof(brokerAccountId), 
+                    "Broker account not found. Validation should be made in Brokerage. " +
```
Comment (KonstantinRyazantsev) [inline src/Api/WebApi/AccountController.cs]: "Broker account not found" is enough ;)

File: src/Api/WebApi/Models/Accounts/BrokerAccountResponse.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Text.Json.Serialization;
+
+namespace Api.WebApi.Models.Accounts
+{
+    public class AccountResponse
+    {
+        [JsonPropertyName("requestId")]
```
Comment (KonstantinRyazantsev) [inline src/Api/WebApi/Models/Accounts/BrokerAccountResponse.cs]: PropertyName -> propertyName already handled by JsonSerializerSettings, so we don't need it here

File: src/Api.Worker/MessageConsumers/AccountActivatedConsumer.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            _accountRepository = accountRepository;
+        }
+
+        public async Task Consume(ConsumeContext<AccountActivated> context)
+        {
+            var message = context.Message;
+
+            var account = await _accountRepository.GetAsync(message.AccountId);
+
+            if (!account.ActivationDateTime.HasValue)
```
Comment (KonstantinRyazantsev) [inline src/Api.Worker/MessageConsumers/AccountActivatedConsumer.cs]: This is not needed. See the comment in BrokerAccountActivatedConsumer

File: src/Api.Common/Domain/Accounts/Account.cs (hunk 1 above)
Comment (OttoVT) [reply src/Api.Common/Domain/Accounts/Account.cs]: this was a copy past that i missed

File: src/Api/WebApi/AccountController.cs (hunk 6 above)
Comment (OttoVT) [reply src/Api/WebApi/AccountController.cs]: this is your message - fix a story, please

Comment (KonstantinRyazantsev) [reply src/Api/WebApi/AccountController.cs]: It's not a message to display to the user. It's just a description for the developer :)

**Pull request merged**